aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -456,6 +457,372 @@ static void upsample_bilinear2d_backward_out_cuda_template(
+  int maxThreadsPerBlock = at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
+  int* maxGridSize = at::cuda::getCurrentDeviceProperties()->maxGridSize;
+  int block_x = std::min<int>(maxThreadsDim[0], at::cuda::warp_size());
+  int grid_x = std::min<int>(maxGridSize[0], ceil_div(output_width, block_x));
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+      input.scalar_type(), "upsample_bilinear2d_out_frame", [&] {
+        using accscalar_t = at::acc_type<scalar_t, true>;
+
+        auto idata = input.packed_accessor64<scalar_t, 4>();
+        auto odata = output.packed_accessor64<scalar_t, 4>();
+
+        const accscalar_t height_scale = area_pixel_compute_scale<accscalar_t>(
+            input_height, output_height, align_corners, scales_h);
+        const accscalar_t width_scale = area_pixel_compute_scale<accscalar_t>(
+            input_width, output_width, align_corners, scales_w);
+
+        // We are using shared memory to store weights wx, wy and a buffer of size wy unique per thread
+        // Let's compute block_y size depending on given height_scale and width_scale
+        // We have the following relationship:
+        // shmem_size / sizeofdtype =
+        //  (width_scale * interp_size + 3) * block_x +   <-- wx allocation
+        //  (height_scale * interp_size + 3) * block_y * (block_x + 1)   <-- wy and buffer allocations
+        // Note: scale * interp_size + 3 is an approximation for ceil( interp_size * 0.5 * scale ) * 2 + 1
+        // See definition of interp_height or interp_width inside the kernel
+
+        int numer = (sharedMemPerBlock * 1.0 / sizeof(scalar_t) - (width_scale * interp_size + 3) * block_x);
+        int denom = (height_scale * interp_size + 3) * (block_x + 1);
+        int block_y = lastPow2((unsigned int) (numer / denom));
+        block_y = std::min<int>(maxThreadsPerBlock / block_x, block_y);
+        const dim3 block(block_x, block_y);
+
+        int grid_y = std::min<int>(maxGridSize[1], ceil_div(output_height, block_y));
+        const dim3 grid(grid_x, grid_y);
+
+        // Compute actual size of required shared memory and verify if we can allocate it
+        // - wx and wy size:
+        size_t weights_per_block =
+            (size_t) (width_scale * interp_size + 3) * block_x + (size_t)(height_scale * interp_size + 3) * block_y;
+        // - buffer size:
+        weights_per_block += (size_t)(height_scale * interp_size + 3) * block_y * block_x;
+        size_t shmem_size = weights_per_block * sizeof(scalar_t);
+        TORCH_CHECK(
+            shmem_size <= sharedMemPerBlock,
+            "Too much shared memory required: ", shmem_size, " vs ", sharedMemPerBlock);
+
+        upsample_gen2d_aa_out_frame<scalar_t, accscalar_t, interp_size>
+            <<<grid,
+               block,
+               shmem_size,
+               stream>>>(height_scale, width_scale, align_corners, idata, odata);
+        C10_CUDA_KERNEL_LAUNCH_CHECK();
+      });
+}
+
+template <int interp_size>
```

Comment:
`interp_size` somewhat obfuscates the fact that it distinguishes between bilinear and bicubic interpolation, maybe add a comment here?